service/route_handlers.go

```diff
@@ -386,6 +386,7 @@ func HandleSamlLogin(w http.ResponseWriter, r *http.Request) {
 
 	if !isWhitelisted(redirectBackBaseValue, s.RedirectWhitelist) {
 		log.Errorf("Cannot redirect to anything other than whitelisted domains and rancher api host")
+		ReturnHTTPError(w, r, http.StatusInternalServerError, "Cannot redirect to anything other than whitelisted domains and rancher api host")
```

Comment:
I think we should return http.StatusForbidden than a 500, internal server error.